atlasdb-client/src/test/java/com/palantir/atlasdb/transaction/impl/ReadTransactionShould.java

```diff
@@ -0,0 +1,165 @@
+                DUMMY_CONSERVATIVE_TABLE,
+                TableMetadataPersistence.SweepStrategy.CONSERVATIVE,
+                DUMMY_THOROUGH_TABLE,
+                TableMetadataPersistence.SweepStrategy.THOROUGH));
+        readTransaction = new ReadTransaction(delegateTransaction, sweepStrategies);
+    }
+
+    @Test
+    public void notAllowPuts() {
+        checkThrowsAndNoInteraction(() -> readTransaction.put(
+                DUMMY_CONSERVATIVE_TABLE,
+                ImmutableMap.of(DUMMY_CELL, "value".getBytes())),
+                IllegalArgumentException.class,
+                "is a read only transaction");
+    }
+
+    @Test
+    public void notAllowDeletes() {
+        checkThrowsAndNoInteraction(() -> readTransaction.delete(
+                DUMMY_CONSERVATIVE_TABLE,
+                ImmutableSet.of(DUMMY_CELL)),
+                IllegalArgumentException.class,
+                "is a read only transaction");
+    }
+
+    @Test
+    public void allowGetsOnConservativeTables() {
+        ImmutableSet<Cell> cellToGet = ImmutableSet.of(DUMMY_CELL);
+        readTransaction.get(DUMMY_CONSERVATIVE_TABLE, cellToGet);
+        Mockito.verify(delegateTransaction, times(1)).get(eq(DUMMY_CONSERVATIVE_TABLE), eq(cellToGet));
+    }
+
+    @Test
+    public void notAllowGetsOnThoroughTables() {
+        ImmutableSet<Cell> cellToGet = ImmutableSet.of(DUMMY_CELL);
+        checkThrowsAndNoInteraction(() -> readTransaction.get(DUMMY_THOROUGH_TABLE, cellToGet),
+                IllegalStateException.class,
+                "Cannot read");
+    }
+
+    @Test
+    public void notAllowSimpleGets() throws IllegalAccessException {
+        Method[] declaredMethods = ReadTransaction.class.getDeclaredMethods();
+
+        for (Method m : declaredMethods) {
+            if (simpleGets.containsKey(m.getName())) {
+                checkThrowsAndNoInteraction(() -> {
+                            try {
+                                m.invoke(readTransaction, simpleGets.get(m.getName()));
+                            } catch (InvocationTargetException e) {
+                                Throwables.throwIfInstance(e.getCause(), IllegalStateException.class);
+                            } catch (IllegalAccessException e) {
+                                Throwables.throwUncheckedException(e);
+                            }
+                        },
+                        IllegalStateException.class,
+                        "Cannot read");
+            }
+        }
+    }
+
+    @Test
+    public void notAllowColumnRangeGets() {
+        checkThrowsAndNoInteraction(() -> readTransaction.getRowsColumnRange(
+                DUMMY_THOROUGH_TABLE,
+                ImmutableList.of(EMPTY_BYTES),
+                BatchColumnRangeSelection.create(EMPTY_BYTES, EMPTY_BYTES, 1)),
+                IllegalStateException.class,
+                "Cannot read");
+    }
+
+    @Test
+    public void notAllowColumnRangeGets2() {
```

Comment:
nit: perhaps `notAllowBatchColumnRangeGets` and `notAllowColumnRangeGets` would be better names.